### PR TITLE
fix(tcp): remove correct listener in`Transport::remove_listener`

### PR DESCRIPTION
**Pull request by jsantell**
## Description

`libp2p_tcp::Transport::remove_listener` previously removed the first listener that *did not match* the provided `ListenerId`. This small fix brings it inline with other implementations.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: transports/tcp/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # 0.39.0 [unreleased]
 
 - Update to `libp2p-core` `v0.39.0`.
+- Fix `Transport::remove_listener` [PR 3387](https://github.com/libp2p/rust-libp2p/pull/3387).
```
Comment (thomaseizinger) [inline transports/tcp/CHANGELOG.md]: Please use a link reference like in the other entries.

Also, can you include here what is actually being fixed?

Like:

```suggestion
- Fix a bug where we removed any other listener in `Transport::remove_listener` except for the one with the provided `ListenerId`. See [PR 3387].
```

**Review by thomaseizinger — COMMENTED** [review]
Nice catch! Thanks for adding a test too!

Just one request about the changelog entry!

**Comment by jsantell** [comment]
Thanks for the speedy feedback @thomaseizinger! Updated the changelog with your suggestion and a PR link

**Review by thomaseizinger — DISMISSED** [review]
LGTM, thanks for the fix!

**Review by thomaseizinger — DISMISSED** [review]
No need to force-push / update the PR unless there are conflicts, the merge bot will handle it.

**Comment by thomaseizinger** [comment]
@mergifyio refresh

**Review by thomaseizinger — DISMISSED** [review]

**Review by mxinden — DISMISSED** [review]
Thank you @jsantell for both debugging and providing the patch!

**Review by mxinden — APPROVED** [review]

**Comment by mxinden** [comment]
I am planning to cut `v0.51.0` in two weeks. In case that is not fast enough, we might be able to do a patch release. Please comment in case that is important for you.

**Comment by jsantell** [comment]
> I am planning to cut `v0.51.0` in two weeks. In case that is not fast enough, we might be able to do a patch release. Please comment in case that is important for you.

Thanks for the merge! `v0.51.0` in two weeks will be plenty fast for us, thank you!

**Comment by cdata** [comment]
Hi @mxinden I work with @jsantell 

We're eagerly anticipating the 0.51.0 release. Do you think that's on schedule to come out this week?

**Comment by mxinden** [comment]
@jsantell and @cdata sorry for the delay here. I was on vacation and caught a cold afterwards. I will kick things off next week.

**Pull request merged**